multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -97,55 +80,27 @@ func NewMemberClusterAnnounceReconciler(client client.Client, scheme *runtime.Sc
 // Reconcile implements cluster status management on the leader cluster
 func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	memberAnnounce := &multiclusterv1alpha1.MemberClusterAnnounce{}
-	err := r.Get(ctx, req.NamespacedName, memberAnnounce)
+	var err error
+	memberID := getIDFromName(req.Name)
+	err = r.Get(ctx, req.NamespacedName, memberAnnounce)
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			// Cannot read the requested resource. Return error, so reconciliation will be retried.
-			return ctrl.Result{}, err
-		}
-		// If MemberClusterAnnounce is deleted, no need to process because RemoveMember must already
-		// be called.
+		// If MemberClusterAnnounce is deleted, no further processing is needed, as cleanup
+		// must have been done when the Finalizer was removed.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	r.mapLock.Lock()
-	// If the member is not found in timerData, it means the member is not added to the ClusterSet or was deleted.
-	if data, ok := r.timerData[common.ClusterID(memberAnnounce.ClusterID)]; ok {
-		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
-		// Reset lastUpdateTime and connectedLeader for this member.
-		data.lastUpdateTime = time.Now()
-		if len(memberAnnounce.LeaderClusterID) == 0 {
-			data.leaderStatus.connectedLeader = v1.ConditionUnknown
-			data.leaderStatus.message = "Not connected to leader yet"
-			data.leaderStatus.reason = ""
-		} else {
-			data.leaderStatus.connectedLeader = v1.ConditionFalse
-			data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
-				memberAnnounce.ClusterID)
-			data.leaderStatus.reason = ReasonNotLeader
-			// Check whether this local cluster is the leader for this member.
-			clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
-			if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
-				if clusterClaim.Value == memberAnnounce.LeaderClusterID {
-					data.leaderStatus.connectedLeader = v1.ConditionTrue
-					data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
-						memberAnnounce.ClusterID)
-					data.leaderStatus.reason = ReasonConnectedLeader
-				}
-			}
-			// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
-			// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
-			// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
-		}
-	}
-	r.mapLock.Unlock()
+	// Initialize the member status with unknown status when controller restarts.
+	// Let's background process to check and update the connection status.
+	// v1.ConditionUnknown will be ignored if status already exists in the map.
+	r.addOrUpdateMemberStatus(memberID, v1.ConditionUnknown)
```

Comment:
Can we remove this call, and keep only a single call of addOrUpdateMemberStatus(), e.g. at line 112?

Reply:
Fixed.